sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -91,14 +91,25 @@ object SQLExecution {
         var ex: Option[Throwable] = None
         val startTime = System.nanoTime()
         try {
-          sc.listenerBus.post(SparkListenerSQLExecutionStart(
+
+          // queryExecution.executedPlan may throw an exception and made the send event message
+          // failed. we should send SparkListenerSQLExecutionStart event to listener bus when we
+          // catch an exception.
+          val sparkPlanInfo = try {
```

Comment:
This seems to fix a separate issue?

Reply:
We use `SparkListenerSQLExecutionStart` event to add `ExecutionListenerBus.activeQueryExecutionIds` here, if we don't send `SparkListenerSQLExecutionStart` event messages when an exception throw, some tests will be failed.